### DISCUSSION TITLE
tmux + helix + `Alt-[` key not registered

### DISCUSSION THREAD

**Discussion by patrick-kidger** [helix-editor/helix#9795] [Troubleshooting]
I've bound `Alt-[` and `Alt-]` for scrolling through Helix buffers (`goto_previous_buffer` and `goto_next_buffer`). The fundamental issue I'm getting is that when running under macOS->iTerm2->fish->tmux->Helix, then `Alt-[` will not change buffers.

However, curiously:
- `Alt-]` (scrolling the other way) does work, so `Alt` is working fine.
- If I rebind `goto_previous_buffer` to just `[` then that works, so `[` is working fine.
- When running Helix without tmux, then `Alt-[` works, so Helix is working fine.
- When running `fish_key_reader` within tmux then `Alt-[` correctly produces `\e\[`, so tmux is working fine.

It seems to be specifically some combination of all parts of the stack, along with this one particular key-combo, that results in the key combo getting swallowed.

(For the record in iTerm2->Settings->Profiles->Keys, I have already set `Left Option Key: Esc+`, as I know that's sometimes the root cause for issues of this nature. Helix version is `23.10 (98ebeeeb)`; tmux version 3.4.)

This is an odd one, so I thought I'd ask here in case anyone has any ideas?

**Comment by phgz** [comment]
Have you found a workaround? I have the exact same problem.

**Reply by patrick-kidger** [reply]
I have not. I ended up needing to redesign my key bindings instead.

**Reply by phgz** [reply]
I use Wezterm and a workaround I found was to redirect the "Alt-[" in the terminal config to something dummy. And using this dummy mapping in the Helix config. So when I physically press "Alt-[" in Helix it indeed does what I want.

**Reply by phgz** [reply]
For future readers: In the end I found out that I had to add `enable_kitty_keyboard = true` (Wezterm) in my terminal configuration. More info here about Enhanced Keyboard Protocol https://github.com/helix-editor/helix/wiki/Terminal-Support#enhanced-keyboard-protocol.
